pkg/reducestreamer/task_manager.go

```diff
@@ -141,26 +139,22 @@ func (rtm *reduceStreamTaskManager) OutputChannel() <-chan *v1.ReduceResponse {
 
 // WaitAll waits for all the reduceStream tasks to complete.
 func (rtm *reduceStreamTaskManager) WaitAll() {
-	tasks := make([]*reduceStreamTask, 0, len(rtm.tasks))
+	var eofResponse *v1.ReduceResponse
 	for _, task := range rtm.tasks {
-		tasks = append(tasks, task)
-	}
-
-	for _, task := range tasks {
 		<-task.doneCh
+		if eofResponse == nil {
+			eofResponse = task.buildEOFResponse()
+		}
 	}
+	rtm.responseCh <- eofResponse
+
```

Comment:
Is this a behaviour change? Before, each task, if not merged, sends its own EOFResponse to responseCh. After, when receiving an EOF, we loop through the active tasks and only send the last EOFResponse.

Reply:
I think we should include verifying EOF in our unit tests. Currently, we are [not](https://github.com/numaproj/numaflow-go/blob/main/pkg/sessionreducer/service_test.go#L1472).

Reply:
I am approving as this is sort of urgent. We can address the unit test comment as a separate issue: https://github.com/numaproj/numaflow-go/issues/113